app/models/host.rb

```diff
@@ -1939,17 +1939,15 @@ def self.node_types
   end
 
   def self.openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    ems.empty? ? false : Host.where(:ems_id => ems).exists?
+    ManageIQ::Providers::Openstack::InfraManager::Host.count > 0
   end
 
   def self.non_openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where.not(:ems_id => ems).exists?
+    (Host.count - ManageIQ::Providers::Openstack::InfraManager::Host.count) > 0
```

Comment:
this is 2 queries. `Host.count`, `Openstack.count` (only value comes over the wire each)

I really prefer:

```ruby
ems = ManageIQ::Providers::Openstack::InfraManager
Host.where.not(:ems_id => ems).exists?
```

this is a single query, and only 1 value comes over the wire.